expression/builtin_encryption.go

```diff
@@ -234,6 +234,9 @@ func (b *builtinPasswordSig) evalString(row types.Row) (d string, isNull bool, e
 		return "", false, nil
 	}
 
+	// Generate warning as password is deprecated as of MySQL 5.7.6
```

Comment:
please add reference here: https://dev.mysql.com/doc/refman/5.7/en/encryption-functions.html#function_password

In addition, it is better say password function rather than password itself. In my naive opinion, password may be confused people while they read your code.